src/electionguard/api/encrypt_ballot.c

```diff
@@ -113,7 +129,113 @@ void API_EncryptBallot_free(struct register_ballot_message message,
     }
 
     if (tracker_string != NULL)
+    {
         free(tracker_string);
+        tracker_string = NULL;
+    }
+}
+
+bool API_EncryptBallot_soft_delete_file(char *export_path, char *filename)
+{
+    bool ok = true;
+    char *default_prefix = "electionguard_encrypted_ballots-";
+    char *existing_filename = malloc(FILENAME_MAX + 1);
+    if (existing_filename == NULL)
+    {
+        ok = false;
+        return ok;
+    }
+
+    char *soft_delete_filename = malloc(FILENAME_MAX + 1);
+    if (soft_delete_filename == NULL)
+    {
+        ok = false;
+        return ok;
+    }
+
+    ok = generate_filename(export_path, filename, default_prefix, existing_filename);
+
+    ok = generate_unique_filename(export_path, filename, default_prefix, soft_delete_filename);
+
+    uint32_t result = rename(existing_filename, soft_delete_filename);
+
+#ifdef DEBUG_PRINT 	
+   if(result != 0) 
+   {
+      printf("API_EncryptBallot_soft_delete_file: unable to rename the file\n\n");
+   }
+#endif
+
+    free(existing_filename);
+    free(soft_delete_filename);
+
+    return ok;
+}
+
+bool export_ballot(char *export_path, char *filename, char **output_filename, 
```

Comment:
Error Handling convention. It seems like upon allocation of `*output_filename` buffer, we do not free the memory upon failure. This may not be obvious to the caller, and could easily lead to memory leaks. I would recommend freeing the memory in case of failure & set `*output_filename` to null to avoid a double-free error.

Reply:
after looking through this, here are the changes I'd like to make:

1. api documentation explicitly states the caller is responsible for freeing this value
2. on allocation error of the string (oom), return early
3. on all failure cases within the method, free the allocated memory and return false

i think that's probably the most consistent pattern.  Do you see any issues?